src/main/resources/avro/references.avdl

```diff
@@ -27,8 +105,22 @@ record Reference {
   string name;
 
   /**
+  Start and end locations of how this reference fits into the existing graph.
+  May be null only for a primary sequence that is not joined onto any other.
+  Note that the referenceId for start and end may not be the same.
+  */
+  union { null, ReferenceJoinLocation } startJoin, endJoin;
+
+  /**
+  The sequence to insert between startJoin and endJoin. Must be set if sourceURI
+  is null.
+  */
+  union { null, string } sequence = null;
```

Comment:
There is an existing and already defined API for getting bases (`ListReferenceBases` method). Agreed that changing that protocol is an entirely separate discussion so #230 is a good place to have that discussion; I don't think it's relevant to this PR.

This method should be used instead of embedding a potentially very long sequence here (or requiring a user to fetch a FASTA file). It may also make sense to eventually augment that method for the Graph API; seems it would be much more convenient if you could fetch a "GraphRange" via ListReferenceBases, in addition to a linear range. Else resolving the reference bases for a graph path may take several serial RPCs.